webpack/webpack-workspace-resolve-plugin/src/KnownDescriptionFilePlugin.ts

```diff
@@ -0,0 +1,117 @@
+// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
+// See LICENSE in the project root for license information.
+
+import type { Resolver } from 'webpack';
+import type { IPrefixMatch } from '@rushstack/lookup-by-path';
+import type { IResolveContext, WorkspaceLayoutCache } from './WorkspaceLayoutCache';
+
+type ResolveRequest = Parameters<Resolver['hooks']['resolveStep']['call']>[1];
+
+/**
+ * A resolver plugin that optimizes locating the package.json file for a module.
+ */
+export class KnownDescriptionFilePlugin {
+  public readonly source: string;
+  public readonly target: string;
+
+  private readonly _skipForContext: boolean;
+  private readonly _cache: WorkspaceLayoutCache;
+
+  public constructor(cache: WorkspaceLayoutCache, source: string, target: string, skipForContext?: boolean) {
```

Comment:
Add a doc comment explaining what these args are (specifically `source` and `target`)?